tests/components/home_connect/test_number.py

```diff
@@ -0,0 +1,173 @@
+            0.1,
+            "°C",
+        ),
+    ],
+)
+@pytest.mark.usefixtures("bypass_throttle")
+async def test_number_entity_functionality(
+    appliance: Mock,
+    entity_id: str,
+    setting_key: str,
+    bypass_throttle: Generator[None],
+    min_value: int,
+    max_value: int,
+    step_size: float,
+    unit_of_measurement: str,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    get_appliances: MagicMock,
+) -> None:
+    """Test number entity functionality."""
+    appliance.get.side_effect = [
+        {
+            ATTR_CONSTRAINTS: {
+                ATTR_MIN: min_value,
+                ATTR_MAX: max_value,
+                ATTR_STEPSIZE: step_size,
+            },
+            ATTR_UNIT: unit_of_measurement,
+        }
+    ]
+    get_appliances.return_value = [appliance]
+    current_value = min_value
+    appliance.status.update({setting_key: {ATTR_VALUE: current_value}})
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert await integration_setup()
+    assert config_entry.state == ConfigEntryState.LOADED
+    assert hass.states.is_state(entity_id, str(current_value))
+    state = hass.states.get(entity_id)
+    assert state.attributes["min"] == min_value
+    assert state.attributes["max"] == max_value
+    assert state.attributes["step"] == step_size
+    assert state.attributes["unit_of_measurement"] == unit_of_measurement
+
+    new_value = random.randint(min_value + 1, max_value)
+    await hass.services.async_call(
+        NUMBER_DOMAIN,
+        SERVICE_SET_VALUE,
+        {
+            ATTR_ENTITY_ID: entity_id,
+            SERVICE_ATTR_VALUE: new_value,
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    appliance.set_setting.assert_called_once_with(setting_key, new_value)
+
+
+@pytest.mark.parametrize("problematic_appliance", ["Refrigerator"], indirect=True)
+@pytest.mark.parametrize(
+    ("entity_id", "setting_key", "mock_attr"),
+    [
+        (
+            f"{NUMBER_DOMAIN.lower()}.refrigerator_refrigerator_temperature",
+            "Refrigeration.FridgeFreezer.Setting.SetpointTemperatureRefrigerator",
+            "set_setting",
+        ),
+    ],
+)
+@pytest.mark.usefixtures("bypass_throttle")
+async def test_number_entity_error(
+    problematic_appliance: Mock,
+    entity_id: str,
+    setting_key: str,
+    mock_attr: str,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    get_appliances: MagicMock,
+) -> None:
+    """Test number entity error."""
+    get_appliances.return_value = [problematic_appliance]
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    problematic_appliance.status.update({setting_key: {}})
+    assert await integration_setup()
+    assert config_entry.state == ConfigEntryState.LOADED
```

Comment:
```suggestion
    assert config_entry.state is ConfigEntryState.LOADED
```

Reply:
This is present in all the tests for Home Connect, among other integrations such as Azure Data Explorer, Bang & Olufsen, and Tibber.

Reply:
uuuuugh, we can use `is` here since its an enum

Reply:
Sure! What I meant in the previous message is that we should also consider changing those in the future.